src/MenuCache.php

```diff
@@ -82,6 +82,8 @@ public function get_menu( $menu, $args ) {
 		if ( $this->should_cache_menu( $args ) && isset( $args->{$this->key_argument} ) ) {
 			$cached_menu = get_transient( $args->{$this->key_argument} );
 			if ( is_string( $cached_menu ) ) {
+				// We have found what we need, so we don't want any code to run after us (since any changes made here would be impossible to restore)
+				remove_all_filters( 'pre_wp_nav_menu' );
```

Comment:
Now that I thought about it for a while, I really don't think we shuold do this.

What you would want to do is to remove all the callbacks that eventually target the menu that we just cached, right? Removing **all** callbacks, however, will affect any menu - also the ones we didn't have the chance to cache. This means that you would end up with using the first cached menu you can find, and then disable the menu cache altogether.

Or am I missing something?